src/jdk.crypto.cryptoki/share/native/libj2pkcs11/p11_sessmgmt.c

```diff
@@ -367,28 +394,83 @@ JNIEXPORT void JNICALL Java_sun_security_pkcs11_wrapper_PKCS11_C_1SetOperationSt
  *                                      CK_ULONG ulPinLen
  */
 JNIEXPORT void JNICALL Java_sun_security_pkcs11_wrapper_PKCS11_C_1Login
-    (JNIEnv *env, jobject obj, jlong jSessionHandle, jlong jUserType, jcharArray jPin)
+    (JNIEnv *env, jobject obj, jlong jSessionHandle, jlong jUserType,
+     jcharArray jPin)
 {
     CK_SESSION_HANDLE ckSessionHandle;
     CK_USER_TYPE ckUserType;
     CK_CHAR_PTR ckpPinArray = NULL_PTR;
     CK_ULONG ckPinLength;
     CK_RV rv;
+    CK_FUNCTION_LIST_PTR ckpFunctions;
 
-    CK_FUNCTION_LIST_PTR ckpFunctions = getFunctionList(env, obj);
-    if (ckpFunctions == NULL) { return; }
+    ckpFunctions = getFunctionList(env, obj);
+
+    if (ckpFunctions == NULL) {
+        return;
+    }
 
     ckSessionHandle = jLongToCKULong(jSessionHandle);
     ckUserType = jLongToCKULong(jUserType);
     jCharArrayToCKCharArray(env, jPin, &ckpPinArray, &ckPinLength);
     if ((*env)->ExceptionCheck(env)) { return; }
 
-    rv = (*ckpFunctions->C_Login)(ckSessionHandle, ckUserType, ckpPinArray, ckPinLength);
+    rv = (*ckpFunctions->C_Login)(ckSessionHandle, ckUserType, ckpPinArray,
+            ckPinLength);
+    free(ckpPinArray);
+
+    if (ckAssertReturnValueOK(env, rv) != CK_ASSERT_OK) { return; }
+}
+#endif
+
+#ifdef P11_ENABLE_C_LOGINUSER
```

Comment:
Is this by default disabled since the LoginUser is commented out in the java code?